unittests/forked_tests.cpp

```diff
@@ -446,4 +438,26 @@ BOOST_AUTO_TEST_CASE(confirmation) try {
 
 } FC_LOG_AND_RETHROW()
 
+BOOST_AUTO_TEST_CASE( read_modes ) try {
+   tester c;
+   c.produce_block();
+   c.produce_block();
+   auto r = c.create_accounts( {N(dan),N(sam),N(pam)} );
+   c.produce_block();
+   auto res = c.set_producers( {N(dan),N(sam),N(pam)} );
+   c.produce_blocks(200);
+   auto head_block_num = c.control->head_block_num();
+
+   tester head(true, db_read_mode::HEAD);
+   push_blocks(c, head);
+   BOOST_REQUIRE_EQUAL(head_block_num, head.control->fork_db_head_block_num());
+   BOOST_REQUIRE_EQUAL(head_block_num, head.control->head_block_num());
+
+   tester irreversible(true, db_read_mode::IRREVERSIBLE);
+   push_blocks(c, irreversible);
+   BOOST_REQUIRE_EQUAL(head_block_num, irreversible.control->fork_db_head_block_num());
+   BOOST_REQUIRE_EQUAL(head_block_num - 49, irreversible.control->head_block_num());
```

Comment:
Instead of hard coding the offset of 49, this test should get the LIB from `c` right after setting `head_block_num` above and compare that to `irreversible.control->head_block_num()`.